### PR TITLE
Comments for BasicContract grpc DTO

### PR DESCRIPTION
**Pull request by alari**

### CODE REVIEW

File: dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala [hunk 1]
```diff
@@ -81,6 +83,9 @@ object BasicContractCodec {
           def read[T](name: String, f: BasicContract ⇒ T): F[T] =
             Option(f(g)).fold[F[T]](F.raiseError(new IllegalArgumentException(s"Required field not found: $name")))(F.pure)
 
+          def readFromOpt[T](name: String, f: BasicContract ⇒ Option[T]): F[T] =
+            Option(f(g)).flatten.fold[F[T]](F.raiseError(new IllegalArgumentException(s"Required field not found: $name")))(F.pure)
```
Comment (DieMyst) [inline dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala]: You don't need to wrap f(g) with Option here

File: dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala [hunk 2]
```diff
@@ -81,6 +83,9 @@ object BasicContractCodec {
           def read[T](name: String, f: BasicContract ⇒ T): F[T] =
             Option(f(g)).fold[F[T]](F.raiseError(new IllegalArgumentException(s"Required field not found: $name")))(F.pure)
 
+          def readFromOpt[T](name: String, f: BasicContract ⇒ Option[T]): F[T] =
```
Comment (DieMyst) [inline dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala]: This is some sort of 'require' or 'unsafe' method, maybe rename it in this way?

File: dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala (hunk 1 above)
Comment (alari) [reply dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala]: Are you completely sure? GRPC code is not always null safe

File: dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala (hunk 2 above)
Comment (alari) [reply dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala]: Why is it unsafe?

Comment (DieMyst) [reply dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala]: It is like option.getOrElse(throw Exception) but under F, as i see

Comment (alari) [reply dataset/grpc/src/main/scala/fluence/dataset/grpc/BasicContractCodec.scala]: It reads into `F`. I think it's ok that `F` uses `raiseError` effect. We could improve it with `Either` later, but now, having `MonadError` in scope, what's the problem?

**Pull request merged**